cmd/soroban-cli/src/events.rs

```diff
@@ -167,12 +167,11 @@ pub enum OutputFormat {
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
-        if self.start_ledger > self.end_ledger {
-            return Err(Error::InvalidLedgerRange {
-                low: self.start_ledger,
-                high: self.end_ledger,
-            });
-        }
+        let start = match (self.start_ledger, self.cursor.clone()) {
+            (Some(start), _) => rpc::EventStart::Ledger(start),
+            (_, Some(c)) => rpc::EventStart::Cursor(c),
+            _ => return Err(Error::MissingStart),
```

Comment:
If you use [required_unless_present](https://docs.rs/clap/latest/clap/struct.Arg.html#method.required_unless_present), we won't need this error (clap will make sure one of them is present)